File: Stacks/Stack.py
class StackUnderFlowException(Exception):
    def __str__(self):
        return "Stack Is Empty"

class Stack:


    def __init__(self):
        self._top = -1
        self._array = []

    def push(self, element):
        if element != None:
            self._top += 1
            self._array.append(element)

    def pop(self):
        if not self.isEmpty():
            element = self._array[self._top]
            self._top -= 1
            self._array.pop()
            return element
        else:
            raise StackUnderFlowException

    def top(self):
        return self._array[self._top]


    def elements(self):
        return self._top + 1;

    def isEmpty(self):
        return self._top == -1

File: Stacks/Stack.js
class Stack {
	constructor() {
		this.items = [];
		this.top = -1;
	}

	// insert an element at the top of the stack
	push(element) {
		if(element !== null || element !== undefined){
			this.top += 1;
			this.items.push(element);
		}
	}

	// remove an element at the top of the stack
	pop() {
		if(!this.isEmpty()){
			var element = this.peek();
			this.top -= 1;
			this.items.pop();
			return element;
		} else {
			throw "StackUnderflow Exception";
		}
	}

	// peek the top element of the 
	// stack without emoving it
	peek() {
		return this.items[this.top];
	}

	// checks if the stack is empty
	isEmpty() {
		return this.top == -1;
	}

	// returns the size of the stack
	size() {
		return this.top + 1;
	}

	// prints the stack
	print() {
		console.log("Stack\n")
	    for (var i = this.top; i >= 0; i--) 
	        console.log(this.items[i] + " ");

	}
}